geoportailv3/static/js/locationControl.js

```diff
@@ -0,0 +1,166 @@
+/**
+ * @fileoverview This file defines the geolocation control.
+ *
+ */
+
+goog.require('ol.Feature');
+goog.require('ol.FeatureOverlay');
+goog.require('ol.Geolocation');
+goog.require('ol.control.Control');
+
+
+goog.provide('app.LocationControl');
+
+
+
+/**
+ * @constructor
+ * @extends {ol.control.Control}
+ * @param {Object=} opt_options Control options.
```

Comment:
You should use a record type and a typedef instead of `Object` for the control options.